### DISCUSSION TITLE
Problem running custom app from a paper

### DISCUSSION THREAD

**Discussion by mfitzka** [idaholab/moose#23156] [Q&A Getting Started] [answered]
Hello all!

I am a new user to MOOSE as I stumbled upon it regarding a paper (https://www.nature.com/articles/s41467-021-23054-7#Sec22). In their paper, they provided the source code to try the application yourself, which can be found under the following link:

https://gitlab.com/zhangx22/concurrentvascfp

I therefore installed MOOSE and tried to run the application, however I could do the given examples as I always ran into an error saying that their custom kernel TempDiffusion is not an registered object.

Therefore, could somebody try to run this on their MOOSE application, as I am wondering if this is a general code problem or has something to do with my MOOSE application.

Any help would be appreciated.

Thanks,
Markus

**Comment by GiudGiud** [comment, marked as answer]
Hello

I just had a look at the repository. The source code to run these input files is not included and is not part of MOOSE. Maybe some of it is part of BISON which is not open source but I cannot confirm that. 

Instead they have included a pre built binary. If you were to have the exact same environment as them, including compilers, libmesh and petsc versions, moose version etc
Then there is a chance this could run on your machine. 

If you want to run these input files you ll need to contact the authors so they share the source code with you. 

Guillaume

**Comment by mfitzka** [comment]
Thank you for your answer!

I already thought so as looking through the input files mentioned several kernels I could not find even with google searches.
Do you think rebuilding their same environment is promising? It seems rather tedious to me.

Markus

**Reply by GiudGiud** [reply]
It's not promising at all, it would be excessively tedious.
It also looks from the bash file in the directory that they were using older packages so you d have to install old versions only for this purpose
```
# GCC 7.2.0
# MPICH 3.2.1
# PETSc 3.8.3
``
